lib/go/test/flow_idtable_staking_test.go

```diff
@@ -927,6 +944,91 @@ func TestIDTableApprovals(t *testing.T) {
 
 	})
 
+	// Move tokens and start a new staking auction
+	tx = createTxWithTemplateAndAuthorizer(b, templates.GenerateMoveTokensScript(env), idTableAddress)
+
+	signAndSubmit(
+		t, b, tx,
+		[]flow.Address{idTableAddress},
+		[]crypto.Signer{IDTableSigner},
+		false,
+	)
+
+	// The current participant access node is participating with zero stake even though the minimum is 100,
+	// because it was approved by the admin
+	assertRoleCountsEquals(t, b, env, []uint16{0, 0, 0, 0, 1})
+	result = executeScriptAndCheck(t, b, templates.GenerateGetCommittedBalanceScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(accessID))})
+	assertEqual(t, CadenceUFix64("0.0"), result)
+	result = executeScriptAndCheck(t, b, templates.GenerateGetStakedBalanceScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(accessID))})
+	assertEqual(t, CadenceUFix64("0.0"), result)
+	result = executeScriptAndCheck(t, b, templates.GenerateReturnCurrentTableScript(env), nil)
+	idArray = result.(cadence.Array).Values
+	assert.Len(t, idArray, 1)
+	assertEqual(t, CadenceString(accessID), idArray[0])
+
+	t.Run("Adding new stake to a zero-stake access node who is already a participant should not add them to the candidate list", func(t *testing.T) {
+		amountToCommit = 10000000000
+		commitNewTokens(t, b, env,
+			accessAddress,
+			accessSigner,
+			amountToCommit,
+			committed[accessID],
+			false)
+
+		result := executeScriptAndCheck(t, b, templates.GenerateGetCommittedBalanceScript(env), [][]byte{jsoncdc.MustEncode(cadence.String(accessID))})
+		assertEqual(t, CadenceUFix64("100.0"), result)
+
+		// Check that they were not added to the candidate node list
+		// expected candidate node list should be empty
+		candidates := CandidateNodes{
+			collector:    []string{},
+			consensus:    []string{},
+			execution:    []string{},
+			verification: []string{},
+			access:       []string{},
+		}
+
+		assertCandidateNodeListEquals(t, b, env, candidates)
+
+	})
+
+	// Approve josh access node
+	tx = createTxWithTemplateAndAuthorizer(b, templates.GenerateAddApprovedAndLimitsScript(env), idTableAddress)
+
+	err = tx.AddArgument(cadence.NewArray([]cadence.Value{CadenceString(adminID), CadenceString(accessID), CadenceString(joshID)}))
+	require.NoError(t, err)
+
+	signAndSubmit(
+		t, b, tx,
+		[]flow.Address{idTableAddress},
+		[]crypto.Signer{IDTableSigner},
+		false,
+	)
+
+	t.Run("Adding new stake to a non-participant, approved, zero-stake access node should add them to the candidate list", func(t *testing.T) {
```

Comment:
Originally I was thinking this case should result in the node _not_ being added to the candidate list. But `removeAndRefundNode` will now remove the node from the approved-list. So we won't encounter the original problem.

The side effect of this is, approved nodes are subject to the random slot selection just like un-approved nodes. I think this is OK though.

Reply:
yeah, that was intentional. We just had a conversation about this last week with the team and decided that it is ok for now